solr/core/src/java/org/apache/solr/handler/export/ExportWriter.java

```diff
@@ -27,9 +27,14 @@
 import java.io.PrintWriter;
 import java.lang.invoke.MethodHandles;
 import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import java.util.TreeSet;
+import org.apache.commons.io.FilenameUtils;
```

Comment:
This isn't a new dependency in core, this is already used elsewhere today: https://github.com/apache/solr/blob/e45be355e2b0fde5f5c80265f1baade024e25e48/solr/core/src/java/org/apache/solr/search/SolrReturnFields.java#L581

Reply:
Ideally we don't pull in more dependencies than needed. I think its weird to use `FilenameUtils` when we are basically just looking at simple globs. 

There was a push to limit dependencies where possible and commons-io was one that just kept getting used all over the place.

Reply:
I updated to create a GlobPatternUtil class that uses Java NIO file pattern matching to avoid using an additional library or having to implement our own glob pattern matching.